sdk/loadtesting/azure-developer-loadtesting/src/test/java/com/azure/developer/loadtesting/LoadTestAdministrationAsyncTests.java

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.developer.loadtesting;
+
+import com.azure.core.http.rest.PagedFlux;
+import com.azure.core.http.rest.RequestOptions;
+import com.azure.core.http.rest.Response;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.polling.AsyncPollResponse;
+import com.azure.core.util.polling.LongRunningOperationStatus;
+import com.azure.core.util.polling.PollerFlux;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+import java.io.File;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public final class LoadTestAdministrationAsyncTests extends LoadTestingClientTestBase {
+
+    // Helpers
+
+    private Map<String, Object> getTestBodyFromDict() {
+        Map<String, Object> testMap = new HashMap<String, Object>();
+        testMap.put("displayName", "Java SDK Sample Test Async");
+        testMap.put("description", "Sample Test Async");
+
+        Map<String, Object> loadTestConfigMap = new HashMap<String, Object>();
+        loadTestConfigMap.put("engineInstances", 1);
+        testMap.put("loadTestConfiguration", loadTestConfigMap);
+
+        Map<String, Object> envVarMap = new HashMap<String, Object>();
+        envVarMap.put("threads_per_engine", 1);
+        envVarMap.put("ramp_up_time", 0);
+        envVarMap.put("duration_in_sec", 10);
+        envVarMap.put("domain", "azure.microsoft.com");
+        envVarMap.put("protocol", "https");
+        testMap.put("environmentVariables", envVarMap);
+
+        return testMap;
+    }
+
+    private BinaryData getFileBodyFromResource(String fileName) {
+        URL url = LoadTestAdministrationAsyncTests.class.getClassLoader().getResource(fileName);
+        return BinaryData.fromFile(new File(url.getPath()).toPath());
+    }
+
+    // Puts and Patches
+
+    @Test
+    @Order(1)
+    public void createOrUpdateTest() {
+        BinaryData body = BinaryData.fromObject(getTestBodyFromDict());
+        Mono<Response<BinaryData>> monoResponse = adminBuilder.buildAsyncClient().createOrUpdateTestWithResponse(newTestIdAsync, body, null);
```

Comment:
Enable [AssertingClient Testing](https://github.com/Azure/azure-sdk-for-java/pull/33177/files#diff-53439b928744c1bab1221881a598ddc21919338f3ad896811f32c0213a207a56R69-R83), this makes sure sync call API's are running in completely synchronous call stack and vice-versa for async stack.

Reply:
Added AssertingClient